jumpscale/sals/zos/container.py

```diff
@@ -1,60 +1,71 @@
-import netaddr
-from jumpscale.core.exceptions import Input
-import base58
-from nacl import signing
-from .id import _next_workload_id
-from nacl import public
 import binascii
+from typing import List, Union
+
+import base58
+import netaddr
+from nacl import public, signing
+
 from jumpscale.clients.explorer.models import (
     Container,
     ContainerLogs,
     ContainerNetworkConnection,
     DiskType,
     WorkloadType,
 )
+from jumpscale.core.exceptions import Input
 
 
 class ContainerGenerator:
     def create(
         self,
-        node_id,
-        network_name,
-        ip_address,
-        flist,
-        capacity_pool_id,
-        env={},
-        cpu=1,
-        memory=1024,
-        disk_size=256,
-        disk_type=DiskType.SSD,
-        entrypoint="",
-        interactive=False,
-        secret_env={},
-        public_ipv6=False,
-        storage_url="zdb://hub.grid.tf:9900",
-    ):
-        """Creates and add a container to the reservation
-
-        Args:
-            reservation (jumpscale.clients.explorer.models.TfgridWorkloadsReservation1): reservation object
-            node_id (str): id of the node of the container
-            network_name (str): identifier of the network
-            ip_address (str): container ip address in the network
-            flist (str): flist url to start the container with
-            env (dict, optional): Environment variables to set. Defaults to {}.
-            cpu (int, optional): CPU capacity. Defaults to 1.
-            memory (int, optional): Memory capacity. Defaults to 1024.
-            entrypoint (str, optional): Container init command. Defaults to "".
-            interactive (bool, optional): Specifies interactive contaienr start or not. Defaults to False.
-            secret_env (dict, optional): Secret Environment variables to set. Defaults to {}.
-            public_ipv6 (bool, optional): IPV6 container ip address in the network. Defaults to False.
-            storage_url (str, optional): Server used as storage backend. Defaults to "zdb://hub.grid.tf:9900".
-
-        Raises:
```

Comment:
here we lost the expected exceptions is there a way to include in the docstrings?

Reply:
it's because the function doesn't raise this exception anymore.